src/psycopmlutils/timeseriesflattener/flattened_dataset.py

```diff
@@ -289,30 +302,225 @@ def _validate_processed_arg_dicts(self, arg_dicts: list):
                 f"Didn't generate any features because: {warnings}",
             )
 
-    def _flatten_temporal_values_to_df_wrapper(self, kwargs_dict: Dict) -> DataFrame:
-        """Wrap flatten_temporal_values_to_df with kwargs for multithreading
-        pool.
+    def _get_feature(self, kwargs_dict: Dict) -> DataFrame:
+        """Get features. Either load from cache, or generate if necessary.
 
         Args:
             kwargs_dict (Dict): Dictionary of kwargs
 
         Returns:
             DataFrame: DataFrame generates with create_flattened_df
         """
-        return self.flatten_temporal_values_to_df(
-            prediction_times_with_uuid_df=self.df[
-                [
-                    self.pred_time_uuid_col_name,
-                    self.id_col_name,
-                    self.timestamp_col_name,
-                ]
-            ],
-            id_col_name=self.id_col_name,
-            timestamp_col_name=self.timestamp_col_name,
-            pred_time_uuid_col_name=self.pred_time_uuid_col_name,
-            **kwargs_dict,
+        full_col_str = generate_feature_colname(
```

Comment:
this returns a list if there are multiple values in the dataframe

Reply:
`generate_feature_colname` returns a list? Not quite sure what you mean here :-)

Reply:
I see the confusion - it doesn't in this branch ;) 
Once the text PR is merged though, `generate_feature_colname` will return a list if there are multiple columns in the predictor df. Might want to wait merging until this works with multiple columns as it will break text functionality otherwise (if run with cache)

Reply:
But `generate_feature_colname` doesn't take a df?

Reply:
Ah, I see, it can take a list as `out_col_name`.

Hmm, not sure I agree with not merging. I think getting caching functionality up so we can start iterating on non-NLP models takes priority so I might be able to train a model before Skagen 🥳

Feel free to slide your chair over if you disagree ;-)